CMakeLists.txt

```
@@ -34,22 +34,25 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
 set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
 set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g")
 
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "")
+
+if(NOT CMAKE_CXX_STANDARD MATCHES "14|17")
```

Does it make sense to already add
```suggestion
if(NOT CMAKE_CXX_STANDARD MATCHES "14|17|20")
```
?

---

We are not using any C++20 stuff. I'm not sure about this one:
- By putting it there, it may seem like a C++20 compatible compiler is needed.
- On the other hand, it is optional as indicated, so we are welcoming to use a C++20 feature if someone wants to.

---

My consideration was that you might have a global setting of the cxx standard  (spack allows you to set the cxxstd as a global variant (see https://github.com/key4hep/key4hep-spack/issues/149)), and setting it to 20 would then result in cmake error here, even though it would build fine.

---

Yeah, that makes sense, I'll add it.